src/internal/components/options-list/utils/use-open-state.ts

```diff
@@ -9,10 +9,12 @@ interface OpenStateProps {
 
 export const useOpenState = ({ onOpen, onClose }: OpenStateProps) => {
   const [isOpen, setIsOpen] = useState(false);
+  const [openedWithKeyboard, setOpenedWithKeyboard] = useState(false);
 
-  const openDropdown = () => {
+  const openDropdown = (isKeyboard?: boolean) => {
```

Comment:
Do we have any instances where we do not know the origin of the action? 

I think for the integrity of our code it would be better to have this always explicitly defined

Reply:
good point, will update